packages/core/src/util/IdProvider.ts

```diff
@@ -0,0 +1,9 @@
+/**
+ * Provides unique identifiers for nodes or boxes.
+ *
+ * @see FreUtils.BOX_ID
+ * @see FreUtils.ID
+ */
+export interface IdProvider {
```

Comment:
@sergej-koscejev Maybe you could have a look at https://github.com/LIonWeb-org/lioncore-typescript/blob/main/src/id-generation.ts as well? In some cases you'd want to control the ID generation using data from the node being created.

Reply:
Currently the ID of a FreNode is set in the constructor (see `ConceptUtils.makeConstructor`) and there won't be any data to consider yet. Your suggestion makes sense but looks like it would require more substantial changes.

Reply:
Why would you want to use the contents of a node to define the id? The ID should be completely independent from the content I think.  First because the content may change, secondly, LionWeb will provide a set of ID'd that can be used, which cannot use the contents anyway.